base/util.go

```diff
@@ -1039,7 +1039,25 @@ func InjectJSONProperties(b []byte, kvPairs ...KVPair) (new []byte, err error) {
 
 	kvPairsBytes := make([]KVPairBytes, len(kvPairs))
 	for i, kv := range kvPairs {
-		valBytes, err := JSONMarshal(kv.Val)
+		var valBytes []byte
+		var err error
+
+		switch v := kv.Val.(type) {
+		case int, int8, int16, int32, int64:
+			valBytes = []byte(strconv.Itoa(kv.Val.(int)))
```

Comment:
Since Itoa is just a helper for FormatInt, this would be cleaner (and consistent with Uint) as strconv.FormatInt(int64(v), 10).